apps/rush-lib/src/logic/PackageJsonUpdater.ts

```diff
@@ -306,67 +309,98 @@ export class PackageJsonUpdater {
 
     await installManager.ensureLocalPackageManager();
     let selectedVersion: string | undefined;
+    const rushConfigProjects: RushConfigurationProject[] = this._rushConfiguration.projects;
 
     if (initialSpec && initialSpec !== 'latest') {
       console.log(colors.gray('Finding versions that satisfy the selector: ') + initialSpec);
       console.log();
-      console.log(`Querying registry for all versions of "${packageName}"...`);
 
-      let commandArgs: string[];
-      if (this._rushConfiguration.packageManager === 'yarn') {
-        commandArgs = ['info', packageName, 'versions', '--json'];
-      } else {
-        commandArgs = ['view', packageName, 'versions', '--json'];
+      // determine if the package is a project in the local repository and if the version exists
+      for (const rushConfigProject of rushConfigProjects) {
+        const rushConfigProjectVersion: string = rushConfigProject.packageJson.version;
+        if(rushConfigProject.packageName === packageName && !projects[0].cyclicDependencyProjects.has(packageName)) {
```

Comment:
We don't just want to check the first project in the `projects` array. It should probably check all of them.